### PR TITLE
Setup nightly build

### PR DESCRIPTION
**Pull request by dilin-MS**

### CODE REVIEW

File: scripts/nightlyBuild.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const fs = require('fs');
```
Comment (eriolchan) [inline scripts/nightlyBuild.js]: It seems PROD/RC/nightly build has the similar steps (get aikey, build, package and publish to marketplace), can we share this in one single script? Only the trigger time is different according to pattern of tag.

Comment (dilin-MS) [reply scripts/nightlyBuild.js]: Fixed.

File: .travis.yml [hunk 2]
```diff
@@ -31,12 +32,22 @@ script:
...
-- node scripts/updateResourceLink.js
-- node scripts/genAiKey.js
+- node scripts/updateConfig.js
 - vsce package
 - docker run -ti --rm -v $PWD:/mnt:ro dkhamsing/awesome_bot --allow-dupe --allow-redirect --skip-save-results `ls *.md`
 
+before_deploy:
+  # Set up git user name and tag this commit if nightly build
+  - if [[ $TRAVIS_EVENT_TYPE == "cron" ]]; then
+      git config --local user.name $USER_NAME &&
```
Comment (dooriya) [inline .travis.yml]: Consider using a service account instead of a personal account.

Comment (dooriya) [reply .travis.yml]: [https://github.com/vsciotci](https://github.com/vsciotci)

---
In reply to: [361107048](https://github.com/microsoft/vscode-iot-workbench/pull/884#discussion_r361107048) [](ancestors = 361107048)

File: scripts/updateConfig.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      packageJson.name = testName;
+      packageJson.displayName = testDisplayName;
+      packageJson.publisher = testPublisher;
+      packageJson.aiKey = process.env['TEST_AIKEY'];
+      trimVersionNumer(packageJson);
+
+      delete packageJson.icon;
+
+
+      // Modify extensionId in template files
```
Comment (dooriya) [inline scripts/updateConfig.js]: too many empty lines here

File: .travis.yml (hunk 2 above)
Comment (dilin-MS) [reply .travis.yml]: Done.

File: scripts/updateConfig.js (hunk 3 above)
Comment (dilin-MS) [reply scripts/updateConfig.js]: Fixed.

**Pull request merged**